src/main/java/jenkins/plugins/logstash/persistence/ElasticSearchDao.java

```diff
@@ -104,7 +108,14 @@ public ElasticSearchDao(URI uri, String username, String password) {
       auth = null;
     }
 
-    clientBuilder = factory == null ? HttpClientBuilder.create() : factory;
+    clientBuilder = factory;
+  }
+
+  HttpClientBuilder clientBuilder() {
```

Comment:
Instances of this class need to be thread safe, so I think it would be better to override `readObject()` instead